rskj-core/src/main/java/co/rsk/mine/gas/provider/MinGasPriceProviderFactory.java

```diff
@@ -0,0 +1,44 @@
+package co.rsk.mine.gas.provider;
+
+import co.rsk.config.mining.StableMinGasPriceSystemConfig;
+import co.rsk.mine.gas.provider.onchain.OnChainMinGasPriceProviderFactory;
+import co.rsk.mine.gas.provider.web.WebStableMinGasPriceProviderFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class MinGasPriceProviderFactory {
+
+    private static final Logger logger = LoggerFactory.getLogger("StableMinGasPrice");
+
+    private MinGasPriceProviderFactory() {
+    }
+
+    public static MinGasPriceProvider create(long fixedMinGasPrice, StableMinGasPriceSystemConfig stableMinGasPriceSystemConfig) {
+        FixedMinGasPriceProvider fixedMinGasPriceProvider = new FixedMinGasPriceProvider(fixedMinGasPrice);
+        if (stableMinGasPriceSystemConfig == null) {
+            logger.warn("Could not find stable min gas price system config, using {} provider", fixedMinGasPriceProvider.getType().name());
+            return fixedMinGasPriceProvider;
+        }
+        if (!stableMinGasPriceSystemConfig.isEnabled()) {
```

Comment:
thought: maybe would be good to have logged here as well that the feature isn't enabled.